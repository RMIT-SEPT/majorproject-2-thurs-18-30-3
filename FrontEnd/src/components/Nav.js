import React from 'react';
import { Link } from 'react-router-dom'

import '../containers/App.css';

//Basic Top Navigation Bar
function Nav() {
    return (
        <nav>
            <div className = "logoDiv">
                <h1>AGME</h1>
                <h4>company</h4>
            </div>

            <u1 className = "nav-links">
                <Link to="/services" className = "big-link">
                    <li>services</li>
                </Link>
                <Link to="/about" className = "big-link">
                    <li>about</li>
                </Link>
                <Link to="/services" className = "big-link">
                    <li>bookings</li>
                </Link>
                <Link to="/employees" className = "big-link">
                    <li>employees</li>
                </Link>

                <ul className = "login-links">
                    <Link to = "/login" className = "little-link">
                        <li>LOGIN</li>
                    </Link>
                    <Link to="/create" className = "little-link">
                        <li>CREATE ACCOUNT</li>
                    </Link>
                </ul>
            </u1>
        </nav>
    );
}

export default Nav;