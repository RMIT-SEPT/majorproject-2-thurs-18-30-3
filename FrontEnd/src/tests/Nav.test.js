import React from 'react';

import Nav from '../components/Nav';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';
import moment from 'moment'; 
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import CurrentUser from '../context/CurrentUser'

//Resize function for testing dynamic hamburger list
beforeAll(() => {
    window.resizeTo = function resizeTo(width, height) {
      Object.assign(this, {
        innerWidth: width,
        innerHeight: height,
        outerWidth: width,
        outerHeight: height,
      }).dispatchEvent(new this.Event('resize'))
    }
  })

const setup = () => {

    const container = render(
        <StaticRouter>
            <CurrentUser.Provider value='true'>
                <Nav/>
            </CurrentUser.Provider>
        </StaticRouter>,
    );
    return container;
}

//Snapshot test
test('Test Nav Renders', () => {

    const comp = renderer.create(
        <StaticRouter>
            <CurrentUser.Provider value='true'>
                <Nav/>
            </CurrentUser.Provider>
        </StaticRouter>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
});

test('Test Nav Elements', () => {

    const comp = setup();
    expect(screen.getByRole('heading')).toHaveTextContent('AGME');
    expect(screen.getAllByRole('link').length).toBe(4);

    expect(screen.getByRole('button')).toHaveAttribute("class","profileButton");
});

test('Test Nav Links', () => {

    const comp = setup();

    const links = screen.getAllByRole('link');
    var about = links[0];
    var services = links[1];
    var book = links[2];
    var emps = links[3];
    

    expect(about).toHaveAttribute('href','/about');
    expect(services).toHaveAttribute('href','/services');
    expect(book).toHaveAttribute('href','/bookings');
    expect(emps).toHaveAttribute('href','/employees');


});

//Check that nav displays dynamic hamburger list when resized
test('Test Nav Resize', () => {
    const comp = setup();
    
    window.resizeTo(700, 700);

    expect(screen.getByRole('button')).toHaveAttribute("class","toggle-button");
});