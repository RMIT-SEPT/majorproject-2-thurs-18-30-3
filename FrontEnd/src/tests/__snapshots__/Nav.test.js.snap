// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Nav Renders 1`] = `
<nav
  className="topNav"
  role="navigation"
>
  <h1>
    AGME
  </h1>
  <ul
    className="nav-links"
  >
    <a
      className="big-link"
      href="/about"
      onClick={[Function]}
    >
      <li>
        about
      </li>
    </a>
    <a
      className="big-link"
      href="/services"
      onClick={[Function]}
    >
      <li>
        services
      </li>
    </a>
    <a
      className="big-link"
      href="/bookings"
      onClick={[Function]}
    >
      <li>
        bookings
      </li>
    </a>
    <a
      className="big-link"
      href="/employees"
      onClick={[Function]}
    >
      <li>
        employees
      </li>
    </a>
    <ul
      className="login-links"
      role="group"
    >
      <button
        className="profileButton"
        onClick={[Function]}
      />
    </ul>
  </ul>
</nav>
`;
